addons/im_livechat/models/discuss_channel_member.py

```diff
@@ -41,3 +41,9 @@ def _get_partner_data(self, fields=None):
                 } if self.partner_id.country_id else False
             return data
         return super()._get_partner_data(fields=fields)
+
+    def _allow_sending_seen_infos_to_channel(self):
+        """ Override to allow sending visitors seen infos to live chat channels. """
+        if self.channel_id.channel_type != 'livechat':
+            return super()._allow_sending_seen_infos_to_channel()
+        return bool(self.guest_id)
```

Comment:
guest is not really correct, we want "visitor" (even if portal, I'm not sure if internal)

Reply:
We don't have a reliable information about "operator(s)":
- In the case of a chatbot, the operator is the chatbot itself, then if a human operator joins, it is just added as a member.
- We could check `livechat_channel.user_ids` to determine if a member is a live chat operator, but then again operators leave/join the channel to manage the number of chats.

We could at least replace this by `not self.env.user._is_internal()`, this way both portal and guests would be correct. I believe it is better to miss the information for internal users rather than leaking the seen info of the operator when one of the above scenario occurs.

Reply:
Wasn't there a way with the order of the members?

Reply:
We do something similar to determine the "correspondent": the first member which is not the operator must be the correspondent. In this case, it's a bit tricky. I'd say the operator should be the third member in the case of a chat bot: the first one is the bot itself, the second is the visitor, the third should be the operator after the `forward_operator` step was done.